### ISSUE TITLE
Test cases failing Java CI build checks but passes on IntelliJ

### ISSUE THREAD

**Issue by nitant-p** [nus-cs2103-AY2223S1/forum#404]
I made a PR adding test cases for a command, and they all pass on my PC on IntelliJ (Using Windows 10, Java version: 11.0.13 IntelliJ version: 2022.2.3)

However, all but four test cases pass on the build check for ubuntu. 

I reran the job multiple times and the build report even states 100% of test cases pass.

Does anyone know what the issue could be?

PR is [here](https://github.com/AY2223S1-CS2103T-W08-3/tp/pull/196)

**Comment by damithc** [comment]
@nitant-p It might help if you share here the link to the said PR.

**Comment by Hongyi6328** [comment]
You can try invalidate cache rebuild the project.

**Comment by nitant-p** [comment]
@Hongyi6328 How do I do that?

**Comment by damithc** [comment]
@nitant-p can any of your team members pass the tests of this PR locally? You can ask them to pull the PR branch from your fork and run the tests locally.

**Comment by damithc** [comment]
@nitant-p Another thing you can try is to run those failing tests locally, but one at a time. This might surface any problems caused by dependencies between tests.

**Comment by pyk595** [comment]
Hi @nitant-p , I'm from your senior batch (dropping by in this forum to see if I can help you guys with tech problems). From what you described, it is possible that your local copy works but it is not working on github actions because there is a regression when your code is merged into the remote repository. You can try pulling the latest copy of your team repository and merging it with your current branch and run the test cases again to debug the issue. Hope this helps!

**Comment by nitant-p** [comment]
@damithc I'll ask my teammates to try run the tests locally. I also reran the failing tests individually on my computer, and they pass as well.

**Comment by damithc** [comment]
@nitant-p did you try @pyk595's suggestion above?

**Comment by nitant-p** [comment]
@damithc I tried it a few times before and it still did not work. 

Currently, a new pull request was just merged into our `main` branch and many other test cases (ones that were previously passing) are no longer passing on my local computer. This is now a different problem, working on it now!

**Comment by nitant-p** [comment]
@pyk595 @damithc I pulled the latest copy from our `master` branch and merged with the PR branch and tried again. 

Unfortunately, the same tests are failing on the CI check, but everything passes on my local machine.

**Comment by RichDom2185** [comment]
I cloned your branch as-is (fresh copy) and I can confirm that the tests are failing on my local environment. I'm running on Apple Silicon Mac with the following config:

```zsh
$ java -version
openjdk version "11.0.16.1" 2022-07-19 LTS
OpenJDK Runtime Environment Zulu11.58+23-CA (build 11.0.16.1+1-LTS)
OpenJDK 64-Bit Server VM Zulu11.58+23-CA (build 11.0.16.1+1-LTS, mixed mode)
```

Commands I ran:
```zsh
$ git clone --branch=nitant-test-edit https://github.com/nitant-p/tp.git nitant-p-tp
... (Output truncated for readibility)
$ cd nitant-p-tp
$ ./gradlew test
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :test

EditCommandParserTest > parse_oneFieldSpecified_success() FAILED
    org.opentest4j.AssertionFailedError at EditCommandParserTest.java:275

EditCommandParserTest > parse_allFieldsSpecified_success() FAILED
    org.opentest4j.AssertionFailedError at EditCommandParserTest.java:199

EditCommandParserTest > parse_multipleRepeatedFields_acceptsLast() FAILED
    org.opentest4j.AssertionFailedError at EditCommandParserTest.java:317

EditCommandParserTest > parse_invalidValueFollowedByValidValue_success() FAILED
    org.opentest4j.AssertionFailedError at EditCommandParserTest.java:346

443 tests completed, 4 failed
... (Output truncated for readibility)
```

Are there some files in your local environment that are not pushed to that branch?

**Comment by damithc** [comment]
Thanks for your help in reproducing the error @RichDom2185

> I cloned your branch as-is (fresh copy) and I can confirm that the tests are failing on my local environment.

@nitant-p you can try the same, in case the tests are influenced by a local file, as @RichDom2185 suggested

**Comment by mjgui** [comment]
Not sure why everyone is awake at this hour, but just wanted to chime in that I managed to replicate the bug as well (Windows 11 x64)

@nitant-p in case you're facing trouble replicating the bug, I decided to step through the debugger to take a look and the issue seems to be a miscapitalized string. In particular, in `CommandTestUtil.java` line 72/73, "pm" should be capitalized to "PM"

For future reference, in case you can't replicate test conditions on your local machine, you can consider re-running Github actions using debug mode to get slightly more helpful output. 
<img width="128" alt="image" src="https://user-images.githubusercontent.com/9080115/199574106-e63f01eb-1ef7-4550-a6a6-a7af4a5ea746.png">

If the output is too dense, a quick hack you can also consider is making a dummy branch, modifying the Github Actions workflow on that branch to only run the tests that are failing, then pushing it (which will trigger the workflow). You can add debug outputs to that branch that will print as long as you run the tests with `--debug` (I believe ticking the checkbox above should work as well).

Hope this is helpful!

**Comment by nitant-p** [comment]
@damithc @RichDom2185 @mjgui Thank you all for the help! I am still not sure exactly why the test cases pass on my local machine despite being synced with my team repo's master branch. There are no local files used in the failing test cases.

@mjgui When I stepped through the debugger "pm" is fine and the `assertEquals` returns true. Anyway, I just changed everything to be upper case by default and it now works on the CI check. Thank you for your help!
